src/OpenTelemetry.Extensions/TracerProviderBuilderExtensions.cs

```diff
@@ -45,6 +45,28 @@ public static class TracerProviderBuilderExtensions
 
 #pragma warning disable CA2000 // Dispose objects before losing scope
         return builder.AddProcessor(new AutoFlushActivityProcessor(predicate, timeoutMilliseconds));
+#pragma warning restore CA2000 // Dispose objects before losing scope
+    }
+
+    /// <summary>
+    /// Adds the <see cref="BaggageActivityProcessor"/> to the <see cref="TracerProviderBuilder"/>.
+    /// </summary>
+    /// <param name="builder"><see cref="TracerProviderBuilder"/> to add the <see cref="BaggageActivityProcessor"/> to.</param>
+    /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+    public static TracerProviderBuilder AddBaggageActivityProcessor(
+        this TracerProviderBuilder builder)
+    {
+#if NET6_0_OR_GREATER
+        ArgumentNullException.ThrowIfNull(builder);
```

Comment:
Please check `Guard` class in this repo. It is available also for .NET Framework.

Reply:
Added in https://github.com/open-telemetry/opentelemetry-dotnet-contrib/pull/1659/commits/2a4852ba4bf30e10d36c875393b4bb243ccf52c6